### PR TITLE
New app: Spinbyt

### PR DESCRIPTION
**Pull request by zachlucas**
This is an app to show information about the nearest [Spin](https://www.spin.app) scooters. Takes two inputs, a specific location for which to check for scooters, and the city in which your Spin service is in.

It calls the Spin API that uses the GBFS standard: https://github.com/MobilityData/gbfs/blob/v2.3/gbfs.md

## Screenshots

spinbyt-2

spinbyt-explainer

### CODE REVIEW

**Comment by zachlucas** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by zachlucas** [comment]
@betterengineering Hi! Would be curious to hear y'all's thoughts on this - anything that I can improve or change here? Thanks!

File: apps/spinbyt/spinbyt.star [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+    # Location (Lat/Lon) from configuration:
+    location = config.get("location", DEFAULT_LOCATION)
+    loc = json.decode(location)
+
+    # Simple scooter image:
+    scooter_image = base64.decode("iVBORw0KGgoAAAANSUhEUgAAABwAAAAcCAYAAAByDd+UAAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAHKADAAQAAAABAAAAHAAAAABkvfSiAAABO0lEQVRIDe2V0Q3CMAxEA2IMviokGIYdugd8lIpJwlR8dQNmKFzFBcdOWkErQIhIVZzkzi9OIXXuU8173+J5K/8d0BkrSlVXlmVYp25sn0yo4VOCk8C2Wnfv8lTsooKmAM+jjGogATJWsqeGBtjemjucQxKC9DEHwdgAwK7dj3VsPu03Fc5uzR02WjfZ2AAny5xJ9AdmDub16d8/0sXrh2OducuBlwccTwNzSS3+MQMPoVkgLm7v3agPMiCAbffHQDdACE5heTjgzqmsqiraJNYvy1WYiz5PgMHYNA39riiKLtaJg0AEGoaluq67Kuk3QAljLpgY53rCpDY1F/0PCYOJDwA05mB6vk8fAaWxzyR1qVhWqdfNj4aCPhM1socem9Qb1XnMu9EGJNUmCdKx9Kd8BogEQyYN+erxFYZIkUf9Flp2AAAAAElFTkSuQmCC")
+
+    # Make a GET request
+    rep = http.get(url)
```
Comment (matslina) [inline apps/spinbyt/spinbyt.star:43]: We'll need a bit of caching around this call to make sure we're not hammering the API too much. Every Tidbyt running the app will trigger one execution of `main()` each time the app is on display, so caching is essential to keep request rates down.

**Review by matslina — CHANGES_REQUESTED** [review]
Hey @zachlucas. Thanks for the PR! And our apologies for the delay. We usually try to merge PRs once a week, but the last 2 weeks have been especially hectic and we're behind!

This looks great. Just the one comment on caching and I think we're good to go.

File: apps/spinbyt/spinbyt.star (hunk 1 above)
[the remark this answers is not in the document]
Comment (zachlucas) [reply apps/spinbyt/spinbyt.star:43]: Makes sense, I see caching recommendations [here](https://tidbyt.dev/docs/build/crypto-tracker#caching), I'll try that approach. Thanks

[the remark this answers is not in the document]
Comment (zachlucas) [reply apps/spinbyt/spinbyt.star]: Screenshot 2022-11-14 at 9 56 38 AM

Looks good!

**Review by matslina — APPROVED** [review]

**Pull request merged**